app/src/components/modal/modal_connect_wallet/index.tsx

```diff
@@ -204,29 +246,34 @@ export const ModalConnectWallet = (props: Props) => {
   const isConnectingToWallet = connectingToMetamask || connectingToWalletConnect || connectingToAuthereum
   let connectingText = `Connecting to wallet`
   if (connectingToMetamask) {
-    connectingText = 'Waiting for Approval on Metamask'
+    connectingText = 'Requesting permission on Metamask'
   }
   if (connectingToWalletConnect) {
     connectingText = 'Opening QR for Wallet Connect'
   }
 
-  const windowObj: any = window
-
   const disableMetamask: boolean = !isMetamaskEnabled || false
   const disableWalletConnect = false
-  const disableAuthereum =
-    XDAI_NETWORKS.includes(windowObj.ethereum && windowObj.ethereum.chainId) ||
-    (networkIds as any)[location.host.split('.')[0].toUpperCase()] === networkIds.XDAI
+
+  React.useEffect(() => {
+    Modal.setAppElement('#root')
+  }, [])
 
   return (
     <>
-      <ModalWrapper
+      <Modal
         isOpen={!context.account && isOpen}
         onRequestClose={onClickCloseButton}
         shouldCloseOnOverlayClick={!isConnectingToWallet}
-        title={connectingToMetamask ? 'Connecting...' : 'Connect a Wallet'}
+        style={theme.connectWalletModal}
       >
         <ContentWrapper>
+          <ModalNavigation>
+            {isConnectingToWallet ? <IconArrowBack hoverEffect={true} onClick={resetEverything} /> : <div></div>}
+            <IconClose hoverEffect={true} onClick={onClickCloseButton} />
+          </ModalNavigation>
+          <IconOmen />
+          <HeaderText>{isConnectingToWallet ? 'Unlock Wallet' : 'Connect a Wallet'}</HeaderText>
           {isConnectingToWallet ? (
             <>
               <Spinner />
```

Comment:
Maybe we can give this spinner big property since it's  smaller then in designs...